llvm/test/tools/sycl-post-link/omit_kernel_args.ll

```diff
@@ -0,0 +1,44 @@
+; This test checks that the post-link tool generates correct kernel parameter
+; optimization info into a property file if the source IR contained
+; corresponding metadata.
+;
+; RUN: sycl-post-link -emit-param-info -S %s -o %t.files.table
+; RUN: FileCheck %s -input-file=%t.files.table --check-prefixes CHECK-TABLE
+; RUN: FileCheck %s -input-file=%t.files_0.prop --match-full-lines --check-prefixes CHECK-PROP
+
+target triple = "spir64-unknown-unknown-sycldevice"
+
+define weak_odr spir_kernel void @SpirKernel1(float %arg1) !spir_kernel_omit_args !0 {
+  call void @foo(float %arg1)
+  ret void
+}
+
+define weak_odr spir_kernel void @SpirKernel2(i8 %arg1, i8 %arg2, i8 %arg3) !spir_kernel_omit_args !1 {
+  call void @bar(i8 %arg1)
+  call void @bar(i8 %arg2)
+  call void @bar(i8 %arg3)
+  ret void
+}
+
+declare void @foo(float)
+declare void @bar(i8)
+
+; // HEX: 0x01 (1 byte)
+; // BIN: 01 - 2 arguments total, 1 remains
+; // NOTE: In the metadata string below, unlike bitmask above, least significant
+; //       element goes first
+!0 = !{i1 true, i1 false}
+
+; // HEX: 0x7E 0x06 (2 bytes)
+; // BIN: 110 01111110 - 11 arguments total, 3 remain
```

Comment:
Does it mean that original kernel had 11 arguments, but only 3 of them are used?

Reply:
yes

Reply:
It is just a bit strange, because I would expect "true" or "1" value for non-dead argument.

Reply:
Agree. But this is how the DeadArgElimination defines it